packages/mobile/src/send/SendAmount/useTransactionCallbacks.ts

```diff
@@ -0,0 +1,207 @@
+import BigNumber from 'bignumber.js'
+import { useCallback, useMemo } from 'react'
+import { useDispatch } from 'react-redux'
+import { hideAlert, showError } from 'src/alert/actions'
+import { RequestEvents, SendEvents } from 'src/analytics/Events'
+import { SendOrigin } from 'src/analytics/types'
+import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
+import { TokenTransactionType } from 'src/apollo/types'
+import { ErrorMessages } from 'src/app/ErrorMessages'
+import {
+  ALERT_BANNER_DURATION,
+  DEFAULT_DAILY_PAYMENT_LIMIT_CUSD,
+  DOLLAR_TRANSACTION_MIN_AMOUNT,
+} from 'src/config'
+import { getFeeEstimateDollars } from 'src/fees/selectors'
+import { AddressValidationType, secureSendPhoneNumberMappingSelector } from 'src/identity/reducer'
+import { getAddressValidationType } from 'src/identity/secureSend'
+import { RecipientVerificationStatus } from 'src/identity/types'
+import { convertToMaxSupportedPrecision } from 'src/localCurrency/convert'
+import {
+  useConvertBetweenCurrencies,
+  useLocalAmountInStableCurrency,
+  useStableCurrencyAmountInLocal,
+} from 'src/localCurrency/hooks'
+import {
+  getLocalCurrencyCode,
+  localCurrencyExchangeRateSelector,
+} from 'src/localCurrency/selectors'
+import { navigate } from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import { useRecipientVerificationStatus } from 'src/recipients/hooks'
+import { Recipient } from 'src/recipients/recipient'
+import useSelector from 'src/redux/useSelector'
+import { updateLastUsedCurrency } from 'src/send/actions'
+import { TransactionDataInput } from 'src/send/SendAmount'
+import { getFeeType, useDailyTransferLimitValidator } from 'src/send/utils'
+import { useCurrencyBalance } from 'src/stableToken/hooks'
+import { Currency } from 'src/utils/currencies'
+
+interface Props {
+  recipient: Recipient
+  localAmount: BigNumber
+  transferCurrency: Currency
+  origin: SendOrigin
+  isFromScan: boolean
+}
+
+function useTransactionCallbacks({
+  recipient,
+  localAmount: approximateLocalAmount,
+  transferCurrency,
+  origin,
+  isFromScan,
+}: Props) {
+  const localAmount = useMemo(() => convertToMaxSupportedPrecision(approximateLocalAmount), [
+    approximateLocalAmount,
+  ])
+  const localCurrencyCode = useSelector(getLocalCurrencyCode)
+  const localCurrencyExchangeRate = useSelector(localCurrencyExchangeRateSelector)[transferCurrency]
+  const recipientVerificationStatus = useRecipientVerificationStatus(recipient)
+  const stableBalance = useCurrencyBalance(transferCurrency)
+  const amountInStableCurrency =
+    useLocalAmountInStableCurrency(localAmount, transferCurrency) ?? new BigNumber(0) // TODO: Handle errors better
+
+  const dispatch = useDispatch()
+
+  const getTransactionData = useCallback(
+    (type: TokenTransactionType): TransactionDataInput => ({
+      recipient,
+      amount: amountInStableCurrency ?? new BigNumber(0), // TODO: Handle errors better
```

Comment:
How do we want to handle this better?

Reply:
I made it so that if we haven't fetched the balances or exchange rates we show a connection error when pressing the `Review` button